akvo/rest/viewsets.py

```diff
@@ -22,3 +22,13 @@ class BaseRSRViewSet(viewsets.ModelViewSet):
     permission_classes = (DjangoObjectPermissions, )
     filter_backends = (filters.DjangoFilterBackend, filters.OrderingFilter, )
     ordering_fields = '__all__'
+
```

Comment:
I think you can eliminate the need for the PublicRSRViewSet entirely by adding the BaseRSRViewSet.get_queryset instead:

```python
    def get_queryset(self):
        """Filter out any private projects."""
        for related_obj in self.queryset.model._meta.get_all_related_objects():
            if related_obj.model == Project:
                self.queryset = self.queryset.filter(project__is_private=False)
                break
        return super(BaseRSRViewSet, self).get_queryset()
```

NOTE: this is untested code!